source/common/upstream/eds.cc

```diff
@@ -22,16 +24,21 @@ EdsClusterImpl::EdsClusterImpl(const Json::Object& config, Runtime::Loader& runt
 
 void EdsClusterImpl::initialize() { subscription_->start({cluster_name_}, *this); }
 
-bool EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
+void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
   std::vector<HostSharedPtr> new_hosts;
   if (resources.size() != 1) {
-    ENVOY_LOG(warn, "Unexpected EDS resource length: {}", resources.size());
-    return false;
+    // We need to allow server startup to continue, even if we have a bad
+    // config.
+    runInitializeCallbackIfAny();
```

Comment:
I'm almost persuaded. Would we run `onConfigUpdateComplete()` in scenarios where the config update fails inside the `Subscription` implementation and never even makes it to `onConfigUpdate()` as well? If so, I can see the interface level utility here.